protoutils/messages.go

```diff
@@ -22,19 +22,23 @@ import (
 
 // ResourceNameToProto converts a resource.Name to its proto counterpart.
 func ResourceNameToProto(name resource.Name) *commonpb.ResourceName {
-	return &commonpb.ResourceName{
+	rName := &commonpb.ResourceName{
 		Namespace: string(name.API.Type.Namespace),
 		Type:      name.API.Type.Name,
 		Subtype:   name.API.SubtypeName,
 		Name:      name.ShortName(),
 	}
+	if name.MachinePartID != "" {
+		rName.MachinePartId = &name.MachinePartID
+	}
+	return rName
 }
 
 // ResourceNameFromProto converts a proto ResourceName to its rdk counterpart.
 func ResourceNameFromProto(name *commonpb.ResourceName) resource.Name {
-	return resource.NewName(
+	return resource.NewNameWithPartID(
 		resource.APINamespace(name.Namespace).WithType(name.Type).WithSubtype(name.Subtype),
-		name.Name,
+		name.Name, name.GetMachinePartId(),
```

Comment:
[q] will this a `*string` since this field is `optional`? 

i guess i'll have my answer once we merge the API change...

Reply:
ah wait, you have the API changes locally - interesting, so unset `optional` string fields are still just blank strings?

Reply:
it's a string
```
func (x *ResourceName) GetMachinePartId() string {
	if x != nil && x.MachinePartId != nil {
		return *x.MachinePartId
	}
	return ""
}
```